### PR TITLE
Move the HTML `canvas()` graphics device into the webr package, reimplement using `OffscreenCanvas`

### PR DESCRIPTION
**Pull request by georgestagg**
Now that Emscripten supports using `EM_ASM` in `SIDE_MODULE`s, and all the major browsers now support `OffscreenCanvas`, it makes sense to draw R graphics on the worker thread and transfer an `ImageBitmap` back to the main thread, rather than sending the raw Canvas API commands over the channel.

This has several major benefits,

* Much improved plotting performance, this method is orders of magnitude faster.
* Using `EM_ASM`, numerical values (e.g. font metrics from `measureText()`) can be returned from the Canvas API and acted upon in the graphics device. This leads to much more accurate text placement. Previously we were using estimated metrics for a specific family of sans-serif font.
* International glyphs and features such as RTL text rendering now work _much_ better since text no longer needs to be escaped and encoded for transfer to the main thread.
* We no longer rely on JS `eval()/Function()` to draw graphics.

With this new implementation, users can also name arbitrary fonts from their machine and the browser will handle the rest. The fonts do not need to be uploaded to the webR VFS like when using Cairo bitmap graphics. I've attached[1] an example showing a plot in the typeface Open Dyslexic, which demonstrates some accessibility benefits to this feature.

The downside is that only very modern browsers support `OffscreenCanvas`. However, we are already at the bleeding edge for Wasm exceptions support in any case, and the latest version of the major browsers do support `OffscreenCanvas`. For slightly older Safari we still have `png()` Cairo bitmap graphics as an alternative.

This PR also moves the code from a patch file to R's source code into a flat file in the webR support package. This should help a lot with maintaining the code.

[1] [Plot example](https://github.com/r-wasm/webr/assets/982548/7e9c3986-c949-4d09-bfc2-138728e1fce3)

### CODE REVIEW

**Review by lionel- — APPROVED** [review]
This looks great to me but if you need a thorough review of the device code then @thomasp85 will be able to help better than I can

File: packages/webr/R/canvas.R [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#' Graphics device for webR HTML canvas element
+#'
+#' A graphics device that generates HTML canvas API calls and executes them on
+#' the worker thread using a JavaScript `OffscreenCanvas`. Once drawing has
+#' finished, a JavaScript `ImageBitmap` object is transmitted to the main webR
+#' thread for display. The resulting messages are of type `canvasImage`.
```
[the remark this answers is not in the document]
Comment (georgestagg) [reply packages/webr/R/canvas.R:6]: Ah, bad wording on my part. The messages are of type `Message`, like other webR output messages. The sentence is referring to the fact that the property named `type` on the object is set to `'canvasImage'` in the output. I'll reword this to something like,

"The resulting webR output messages are of type `Message` with the `type` property set to `'canvasImage'` and the `data` property set to contain the image data, `{ image: ImageBitmap }`".

**Comment by georgestagg** [comment]
Thanks! I think just a light look at the device code is fine for now. Other than some additional font support, the drawing logic is basically the same as before and based on the existing R [canvas](https://www.rforge.net/canvas/) package. Here I'm mostly just concerned with how the resulting Canvas API calls are executed, rather than the calls themselves.
